crates/flux-middle/src/rty/mod.rs

```diff
@@ -1195,6 +1195,7 @@ pub struct VariantSig {
     pub args: GenericArgs,
     pub fields: List<Ty>,
     pub idx: Expr,
+    pub invariants: Vec<Invariant>,
```

Comment:
Let's make this a list of expressions, i.e., store the invariants already applied. Also, let's call it `requires`, as in a list of predicates that need to be true to construct the variant.

```suggestion
    pub requires: List<Expr>,
```